codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/XMLProtocolUtils.java

```diff
@@ -275,8 +275,13 @@ public static void writeXmlErrorMessageCodeDeserializer(ProtocolGenerator.Genera
 
         if (requiresS3Customization(service)) {
             writer.addUseImports(AwsCustomGoDependency.S3_SHARED_CUSTOMIZATION);
-            writer.write("errorComponents, err := s3shared.GetErrorResponseComponents(errorBody)");
+            String getErrorComponents = String.format("errorComponents, err := "
+                            + "s3shared.GetErrorResponseComponents(errorBody, response.StatusCode, %s)",
+                    isS3Service(service));
+            writer.write(getErrorComponents);
```

Comment:
This might be a good case where having two different functions would be better than an additional `bool` parameter. `s3control` can use the "generic" version of this function, where `s3` could use the one that wraps the "generic" and adds the additional behavior.

This would result in a if/else here not to bad, but have the benefit of the error functions be specific to their use cases. Individual functions also lets us more finely modify the use cases separately too.